app/services/sort_and_filter_service.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+# This service handles sort and filtering submissions.
+class SortAndFilterService
```

Comment:
nit: could be named more specific e.g.
```suggestion
class SubmissionsSortAndFilterService
```